packages/data/addon/adapters/elide-metadata.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import EmberObject from '@ember/object';
+import { queryManager } from 'ember-apollo-client';
+import GQLQueries from 'navi-data/gql/metadata-queries';
+import { assert } from '@ember/debug';
+
+export default class GraphQLMetadataAdapter extends EmberObject {
+  /**
+   * @property {Object} apollo - apollo client query manager using the overriden metadata service
+   */
+  @queryManager({ service: 'navi-metadata-apollo' })
+  apollo;
+
+  /**
+   * @method fetchAll
+   * @param {String} type
+   * @param {Object} options
+   * @returns {Object} JSON response
+   */
+  fetchAll(type, options) {
+    return this.fetchMetadata(type, undefined, options);
+  }
+
+  /**
+   * @method fetchMetadata
+   * @param {String} type
+   * @param {String} id
+   * @param {Object} options
+   * @returns {Object} JSON response for type
+   */
+  async fetchMetadata(type, id, options = {}) {
```

Comment:
Maybe we can clean up the general adapter interface long term. `fetchAll` & `fetchById` seem like better names.

Reply:
I agree. I just wanted to get what we had close-sourced into the open-source repo first. It's a big PR so I didn't want to fill it with lots of changes that would make it harder to review.

Reply:
Yeah, just a note for future work around navi-data